app/controllers/stages_controller.rb

```diff
@@ -142,6 +141,9 @@ def stage_permitted_params
         deploy_group_ids: [],
         command_ids: []
       }
-    ] + Samson::Hooks.fire(:stage_permitted_params).flatten
+    ].tap do |params|
+      params << :no_code_deployed if current_user.admin?
+      params << Samson::Hooks.fire(:stage_permitted_params)
```

Comment:
- only `stage_permitted_params` needs the flatten
 - prefer assigning a variable and then adding things to it over `tap`-ing